client/src/components/Search/index.js

```diff
@@ -34,15 +36,52 @@ export default class Search extends Component {
 
   handleSubmit(event) {
     // TODO - hookup to map instance and add marker for given location
-    alert('A query was submitted: ', this.state.value);
     event.preventDefault();
+    axios.post(`${API_SERVER}/search/textsearch`, {
+      input: this.state.value,
+      lat: this.props.position.lat,
+      lng: this.props.position.lng,
+      }).then((response) => { 
+        if(response.data.status == OK_Status)
+        {
+          let [searchResult] = response.data.results;
+          this.onPlaceFound(searchResult);
+        }          
+        else
+          alert(response.data.status);
+      
+      });
+      
   }
 
+  /**
+   * On search input query completion, add a marker to the map at the search result.
+   *
+   * @param {*} placeDetail 
+   */
+  onPlaceFound(placeDetail) {
+    this.props.map.setZoom(16);
+    const location = placeDetail.geometry.location;
+    const userMarker = L.marker(location).addTo(this.props.map)
+    .bindPopup(`<b>${placeDetail.name} </b>${placeDetail.formatted_address}`);
+    this.props.map.setView(location, 16);
+    //ToDO : add to state , this location should be separate to user location
```

Comment:
additionally, we only want one instance of this marker, if its not null, we should remove the current marker from the map and replace with new one per user selection. Currently, each search submit persists on the map.